### DISCUSSION TITLE
error: unrecognized arguments: --execute

### DISCUSSION THREAD

**Discussion by madou-sow** [pyccel/pyccel#1037] [Q&A] [answered]
I understand the failure to execute this basic program
 
1. helloworld.py 
```
#! / usr / bin / python
def main ():
    print ('Hello world!')

if __name__ == '__main__':
    main()
```

2. pyccel helloworld.py --execute
error: unrecognized arguments: --execute

**Comment by EmilyBourne** [comment]
Please try and use [github's code markup](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to denote your code. This makes it much easier to understand your question.

Pyccel does not have an `--execute` argument. However your example can be compiled into a program which can then be executed like a normal C program:
```
pyccel helloworld.py
./helloworld
```

**Comment by madou-sow** [comment]
*

> Please try and use [github's code markup](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to denote your code. This makes it much easier to understand your question.
> 
> Pyccel does not have an `--execute` argument. However your example can be compiled into a program which can then be executed like a normal C program:
> 
> ```
> pyccel helloworld.py
> ./helloworld
> ```

Hello I invite you to consult this document by A. Ratnani (https://buildmedia.readthedocs.org/media/pdf/pyccel/latest/pyccel.pdf) on pages 4, 5 and 6 where he uses this option **--execute**

**Comment by EmilyBourne** [comment]
Hi, thanks for the explanation about where this command came from. I have never seen it before. Perhaps @ratnania can comment?

As you can see this document was last compiled in October 2018 and Pyccel has changed a lot since then. The pdf is hosted on readthedocs.org where we used to have our documentation, however we stopped providing the link to this site as almost none of the information was up to date. The replacement documentation is still being written in part, but there are already several topics that are accurately described. Please see [here](https://github.com/pyccel/pyccel#user-documentation) for a table of contents

**Comment by EmilyBourne** [comment, marked as answer]
I have just had a hunt through the history and it seems that the `--execute` command was [removed](https://github.com/pyccel/pyccel/commit/04d910ceafa0eeb711e407f944ce22ee4041dbfb#diff-03c8e58b2510fecdb23a0377e4d36f79b6665e8d5791e0f2c0c21fac503a6f8d) in April 2018, before this pdf was generated so I am not sure why it appears in this document. In 2018 Pyccel was still finding its feet, a lot of the functional details were still being worked out, and not everything was fully tested, so there is no guarantee that what the pdf describes is still accurate. I strongly recommend that you use the documentation indicated in Pyccel's [README](https://github.com/pyccel/pyccel/blob/master/README.md) as this refers only to documents which are kept up to date

**Comment by madou-sow** [comment]
> I have just had a hunt through the history and it seems that the `--execute` command was [removed](https://github.com/pyccel/pyccel/commit/04d910ceafa0eeb711e407f944ce22ee4041dbfb#diff-03c8e58b2510fecdb23a0377e4d36f79b6665e8d5791e0f2c0c21fac503a6f8d) in April 2018, before this pdf was generated so I am not sure why it appears in this document. In 2018 Pyccel was still finding its feet, a lot of the functional details were still being worked out, and not everything was fully tested, so there is no guarantee that what the pdf describes is still accurate. I strongly recommend that you use the documentation indicated in Pyccel's [README](https://github.com/pyccel/pyccel/blob/master/README.md) as this refers only to documents which are kept up to date

Thank you sincerly for your collaboration
